lib/monitor/run.js

```diff
@@ -343,15 +364,21 @@ function kill(child, signal, callback) {
     const sig = signal.replace('SIG', '');
     psTree(child.pid, function (err, kids) {
       if (psTree.hasPS) {
-        spawn('kill', ['-s', sig, child.pid].concat(kids))
-          .on('close', callback);
+        spawn('kill', ['-s', sig].concat(kids))
+          .on('close', waitForOldKids.bind(this, kids, function () {
```

Comment:
If we filter out `child.pid` from `kids` here, we can just rely on the `exit` event from `ChildProcess` to detect the root process exit, which avoids the extra polling in the case that there's no extra child processes.  e.g. `kids.filter(p => p !== child.pid)`.

Reply:
I think I know what you're meaning. But some commands won't create any sub processes. Maybe it is esoteric but for example `nodemon --exec  "read VAR"` won't start a subprocess because `read` is a shell builtin command.

Reply:
`read VAR` will start a shell subprocess, although there will be just one subprocess from nodemon's perspective.  But in that case the `exit` event will be sent to us for that process so we don't need to poll for it to exit.

Reply:
Or to clarify what I'm trying to say.  `nodemon --exec "read VAR"` will at least start `/bin/sh` to run the `read` built-in.  So there's still a child process there, but only one (`/bin/sh`).  `nodemon --exec "python foo.py"` runs two subprocesses - `/bin/sh` and `python`.

Reply:
Ok, you think `kids` contains also the `sh` process PID. I thought it doesn't (because of the previous old code in line 346: `..., child.pid].concat(kids)`). I have to clarify what is right here.

Reply:
Hmm actually I might have been confused about this one, I did a quick test and it seems like `pstree` is not including the parent id in the result.

e.g.

```
> child = require('child_process').exec('sleep 300s')
ChildProcess {
> child.pid
22955
> require('pstree.remy')(child.pid, (e, k) => console.log(k))
undefined
> [ 22956 ]

```

Reply:
I deleted some of my erroneous comments about this...